enhancements/api-review/custom-tags-aws.md

```diff
@@ -31,113 +36,325 @@ have the ability to apply user defined tags to many resources created by OpenShi
+
+2) There is no tag with same key present, new user-defined tag is created for the AWS resource. In case of limit reached, a validation error is generated.\
+   For example,\
+   Existing tag for AWS resource = `key_infra1 = value_comp1`\
+   New tag request = `.spec.platformSpec.aws.resourceTags` has `key_infra1 = value_infra1`\
+   Action = A new tag is created for AWS resource.\
+   Final tag set to AWS resource = `key_infra1 = value_infra1`\
+   Event action = An event is generated to notify user about the action status (success/failure) to create tags for the AWS resource.
+
+### Update tags scenarios
+Users can update the user-defined tags by editing `.spec.platformSpec.aws` of the `Infrastructure.config.openshift.io` type. New addition of tags are always appended.
+On update of `resourceTags`, AWS resource is not created or restarted.
+
+In the case where a user-defined tag is specified in the Infrastructure resource and
+1) There is already user-defined tag with the same key and value present for AWS resource, the user-define tag value for the AWS resource will not be updated.\
+   For example,\
+   Existing tag for AWS resource = `key_infra1 = value_update1`\
+   New tag request = `.spec.platformSpec.aws.resourceTags` has `key_infra1 = value_update1`\
+   Action = There is no update for AWS resource.\
+   Final tag set to AWS resource = `key_infra1 = value_update1`\
+   Event action = An event is generated to notify user about the action status (success/failure) to update tags for the AWS resource.
+
+2) There is already user-defined tag with the same key and different value present for AWS resource, the user-define tag value for the AWS resource will be updated.\
+   For example,\
+   Existing tag for AWS resource = `key_infra1 = value_old`\
+   New tag request = `.spec.platformSpec.aws.resourceTags` has `key_infra1 = value_update1`\
+   Action = Existing tag for AWS resource is updated to reflect new value.\
+   Final tag set to AWS resource = `key_infra1 = value_update1`\
+   Event action = An event is generated to notify user about the action status (success/failure) to update tags for the AWS resource.
+
+3) The new user-defined tag request has empty string `""`, the user-defined tag value for the AWS resource will be updated. Please refer to the `Delete Scenarios` for more details on empty value string handling.\
+   For example,\
+   Existing tag for AWS resource = `key_infra1 = value_old`\
+   New tag request = `.spec.platformSpec.aws.resourceTags` has `key_infra1 =`\
+   Action = Existing tag for AWS resource is updated to reflect new value.\
+   Final tag set to AWS resource = `key_infra1 = value_update1`\
+   Event action = An event is generated to notify user about the action status (success/failure) to update tags for the AWS resource. A warning also must be generated about user-defined tag being marked for deletion.
+
+### Delete tags scenarios
```

Comment:
There is no scenario detailing the behavior when a user-defined tag is actually deleted from the `.spec.platformSpec.aws.resourceTags`. The possibility is mentioned later in the "Precedence scenarios" section.

Reply:
This is mentioned in caveats section.